posts/Yoann/2023-06-02-hibernate-search-6-2-0-Beta2.adoc

```diff
@@ -0,0 +1,300 @@
+= Hibernate Search 6.2.0.Beta1 is out
+Yoann Rodière
+:awestruct-tags: [ "Hibernate Search", "Lucene", "Elasticsearch", "Releases" ]
+:awestruct-layout: blog-post
+:hsearch-doc-url-prefix: https://docs.jboss.org/hibernate/search/6.2/reference/en-US/html_single/
+:hsearch-jira-url-prefix: https://hibernate.atlassian.net/browse
+:hsearch-version-family: 6.2
+:hsearch-jira-project-id: 10061
+:hsearch-jira-version-id: 32141
+---
+
+We just published Hibernate Search 6.2.0.Beta1,
+a beta release of the next minor version of Hibernate Search.
+
+This version brings highlighting in the Search DSL,
+indexing plan filters to suspend automatic indexing,
+an Elasticsearch schema export tool,
+mapping annotations for constructor parameters when using `@ProjectionConstructor`,
+and `.missing().lowest()`/`.missing().highest()` options in sorts.
+
+6.2.0.Beta1 also includes many bugfixes and improvements,
+compatibility with Elasticsearch 8.8 and OpenSearch 2.7,
+an upgrade of `-orm6` artifacts to Hibernate ORM 6.2.4.Final,
+and more.
+
+== What's new
+
+[NOTE]
+====
+Hibernate Search 6.2 is still in development:
+some features are still incomplete or may change in a backward-incompatible way.
+====
+
+=== Dependency upgrades
+
+[[orm-version]]
+Hibernate ORM (link:{hsearch-jira-url-prefix}/HSEARCH-4860[HSEARCH-4860])::
+Hibernate Search now depends on Hibernate ORM 6.2.4.Final for -orm6 artifacts.
+[[elasticsearch-version]]
+Elasticsearch (link:{hsearch-jira-url-prefix}/HSEARCH-4857[HSEARCH-4857])::
+The Elasticsearch backend now works with Elasticsearch 8.8
+as well as other versions that were already compatible.
+[[opensearch-version]]
+OpenSearch (link:{hsearch-jira-url-prefix}/HSEARCH-4850[HSEARCH-4850])::
+The Elasticsearch backend now works with OpenSearch 2.7
+as well as other versions that were already compatible.
+[[others-version]]
+Others::
+* link:{hsearch-jira-url-prefix}/HSEARCH-4857[HSEARCH-4857]: Upgrade to Elasticsearch **client** 8.8.0
+
+[[search-highlighting]]
+=== Highlighting
+
+With link:{hsearch-jira-url-prefix}/HSEARCH-2192[HSEARCH-2192], Hibernate Search now offers a new feature in the Search DSL: highlighting.
+
+Highlighting is a projection that returns fragments from full-text fields of matched documents that caused a query match.
+Specific terms that caused the match are "highlighted" with a pair of opening and closing tags.
+It can help a user to quickly identify the information they were searching for on a results page.
+
+For example, one can enable highlighting on a full-text field like this:
+
+[source, JAVA, indent=0]
+----
+@Entity
+@Indexed
+public class Book {
+    @Id
+    private Integer id;
+
+    @FullTextField(analyzer = "english")
+    private String title;
+
+    @FullTextField(analyzer = "english", projectable = Projectable.YES)
```

Comment:
```suggestion
    @FullTextField(analyzer = "english", highlightable = Highlightable.ANY)
```
not sure if we should do it, but I thought that maybe showing that there's a dedicated attribute for highlights here and then the documentation can show all the cases where it works and where it doesn't.